tokio/src/runtime/basic_scheduler.rs

```diff
@@ -117,13 +133,65 @@ where
         self.spawner.spawn(future)
     }
 
-    pub(crate) fn block_on<F>(&mut self, future: F) -> F::Output
-    where
-        F: Future,
-    {
+    pub(crate) fn block_on<F: Future>(&self, future: F) -> F::Output {
+        // If we can steal the dedicated parker then lets block_on that
+        // Otherwise, we'll block_on the future and attempt to steal the
+        // parker later, if we can.
+        if let Some(mut inner) = self.take_inner() {
+            inner.block_on(future)
+        } else {
+            let enter = crate::runtime::enter(false);
+
+            let mut park = ParkThread::new();
```

Comment:
We will want to use the cached variant of this. Otherwise, we will be creating a new parker on each call.